exchanges/account/account.go

```diff
@@ -292,6 +311,78 @@ func (s *Service) Save(incoming *Holdings, creds *Credentials) error {
 	return errs
 }
 
+// Update updates the balance for a specific exchange and credentials
+func (s *Service) Update(exch string, changes []Change, creds *Credentials) error {
+	if exch == "" {
+		return fmt.Errorf("cannot update balance: %w", errExchangeNameUnset)
+	}
+
+	if creds.IsEmpty() {
+		return fmt.Errorf("cannot update balance: %w", errCredentialsAreNil)
+	}
+
+	exch = strings.ToLower(exch)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	accounts, ok := s.exchangeAccounts[exch]
+	if !ok {
+		var err error
+		if accounts, err = s.trackNewAccounts(exch); err != nil {
+			return fmt.Errorf("cannot update balance for %s %w", exch, err)
+		}
+	}
+
+	var subAccounts map[key.SubAccountAsset]currencyBalances
+	subAccounts, ok = accounts.subAccounts[*creds]
+	if !ok {
+		subAccounts = make(map[key.SubAccountAsset]currencyBalances)
+		accounts.subAccounts[*creds] = subAccounts
+	}
+
+	var errs error
+	for _, change := range changes {
+		if !change.AssetType.IsValid() {
```

Comment:
🤷 Should we check change.Account here before anything else?

Reply:
I think it's fine to keep it as is for now